src/kudu/consensus/leader_election.cc

```diff
@@ -958,30 +1018,77 @@ std::pair<bool, bool> FlexibleVoteCounter::IsDynamicQuorumSatisfied() const {
     return presult;
   }
 
+  // candidate_region is expected to be valid, because Raft Consensus
+  // only starts elections in voters which have valid region
+  std::string candidate_region(
+      DetermineRegionForUUID(candidate_uuid_));
+
   // Step 3: Check if last known leader's quorum is satisfied and we directly
   // succeed term.
-  if (election_term_ == last_known_leader.election_term() + 1) {
+  // Since pessimistic quorum is not satisfied, we have to intersect with last
+  // known leader region to guarantee longest log in next LEADER.
+  // However if there is a period of confusion after there was a stable LEADER,
+  // the CANDIDATE might not be able to know which region to intersect with.
+  //
+  // If a CANDIDATE has heard from all voters then it should have confidence
+  // that its crowdsourced last known leader is the correct one. So if there
+  // is no term continuity, it only means that there was no leader during the
+  // period of confusion
+  bool continuity_not_required =
+      FLAGS_crowdsource_last_known_leader && FLAGS_trust_last_leader_entries &&
+      AreAllVotesIn() && !last_known_leader.uuid().empty();
```

Comment:
This is an optimization to not have to go to ComputeElectionResultFromVotingHistory.
When we have all votes from all voters, we should know who the last leader is because
the voters have sent it to us. term == lkl_term + 1 is restrictive. 

If we have lkl in term 2 but a failed election in term 3, 4 , 5  and candidate starting election in term 6
and all voters have told us that there is no known leader beyond term 2, we can assume that
there is no known leader && if we have the requisite votes from them.

Reply:
The whole point of `ComputeElectionResultFromVotingHistory` is to figure out that there is no leader beyond term 2. Why is this change required?